### ISSUE TITLE
Remove old / unused JDK derivations

### ISSUE THREAD

**Issue by jlesquembre** [NixOS/nixpkgs#313208]
## Issue description

Currently, we have multiple JDK flavors:

- OpenJDK
- AdoptOpenJDK
- OracleJDK
- Temurin
- Zulu
- GraalVM
- JetBrains JDK
- Corretto

There might be others as well. Please let me know if I missed anything.

With the goal of simplifying the JDK infrastructure on nixpkgs, I suggest:

- Removing AdoptOpenJDK
- Hiding all the `openjdkXX-bootstrap` packages from the top level

Can we also remove OracleJDK? I'm not sure if it's needed anymore.

I also believe we could remove Zulu if we are able to compile OpenJDK forDarwin.

**Comment by willcohen** [comment]
I think this sounds good. Re Oracle SDK, I personally do not use it, but is there a use case for people who would want to deploy that install via NixOS if they are purchasing support from Oracle? My initial thought is that that'd something which a user of paid software would handle on their own. For anyone using the software free of charge, OpenJDK seems like it should suffice just fine.

**Comment by tomodachi94** [comment]
OracleJDK is gone as of #353043. I know that @emilazy had opinions about which JDKs to keep and which to drop into the abyss.

**Comment by emilazy** [comment]
I believe AdoptOpenJDK is already gone, and the Oracle JDK definitely is. Zulu can indeed go once we have OpenJDK/OpenJFX on Darwin, although we’d lose JDK 8 on `aarch64-darwin`, which I think is acceptable. (Alternatively we could keep Zulu for bootstrap and ditch Temurin.) Graal is its whole own thing so I don’t have an opinion on it although IIRC @thiagokokada was planning on not maintaining it any more. I believe the JetBrains products require their JDK so it should probably be maintained by maintainers of JetBrains products in Nixpkgs. I don’t personally see the point of Corretto since AFAICT it’s just a vanilla OpenJDK build like ours and Temurin but apparently @rollf does :)

**Comment by emilazy** [comment]
Oh and there’s also Semeru. I assume some people want OpenJ9 although presumably we could build our own instead.

**Comment by tomodachi94** [comment]
> I believe AdoptOpenJDK is already gone

Yes, that appears to be the case. We have [`adoptopenjdk-icedtea-web`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/adoptopenjdk-icedtea-web/default.nix), which I'm not sure about.

> I don’t personally see the point of Corretto since AFAICT it’s just a vanilla OpenJDK build like ours and Temurin but apparently @\\rollf does :)

I agree, though I'm curious about Rolf's opinion (I'm not familiar with Corretto).

**Comment by rollf** [comment]
I mentioned my reasoning [here](https://github.com/NixOS/nixpkgs/pull/344544#issuecomment-2375841245). The short version is: if the build infra structure [locks down the jdk](https://docs.gradle.org/current/userguide/toolchains.html#sec:vendors), you can't use an alternative. Whether/Why the build infra *should* lock that down is an entirely different topic :shrug: .

**Comment by tomodachi94** [comment]
> I'm trying to open a Java Web Start file [...]

Have you tried `adoptopenjdk-icedtea-web`? It appears to let you do this. I think you'll need to also pull in `openjfx` if your JWS file uses that.

**Comment by msgilligan** [comment]
From the perspective of getting newer JDKs (e.g. 25) including in Nixpkgs sooner, I prioritized

* openjdk -- the upstream Open Source,  reference build for JDK
* temurin-bin -- popular Open Source pre-built distribution which Nixpkgs uses to bootstrap OpenJDK
* zulu -- popular Open Source pre-built distribution with JavaFX. On Darwin, Nixpkgs aliases `jdk` to `zulu`
* graalvm-ce -- Open Source "community edition" of GraalVM that provides the `native-image` compiler
* graalvm-oracle -- non-free GraalVM, but provides binaries during Early Access


From https://github.com/NixOS/nixpkgs/issues/430389#issuecomment-3304698501

**Comment by msgilligan** [comment]
Note: A Nixpkgs JDK "Dashboard" would be helpful for tracking these JDKs: https://github.com/NixOS/nixpkgs/issues/425860
